### PR TITLE
Rc3

### PR DESCRIPTION
**Pull request by nmsimons**
This updates the examples to use rc3. Although it currently using a prerelease version. I won't merge until I change that but I would like a review in the meantime.

### CODE REVIEW

File: brainstorm/src/react/noteux.tsx [hunk 1]
```diff
@@ -242,13 +240,18 @@ function NoteToolbar(props: { note: Note; clientId: string; notes: Items }): JSX
...
-		accept: dragType.NOTE,
+		accept: [dragType.NOTE, dragType.GROUP],
 		collect: (monitor) => ({
 			isActive: monitor.canDrop() && monitor.isOver(),
 		}),
+		canDrop: (item) => {
+			if (item instanceof Note) return true;
+			if (item instanceof Group && !Tree.contains(item, props.parent)) return true;
+			return false;
+		},
```
[the remark this answers is not in the document]
Comment (ChumpChief) [reply brainstorm/src/react/noteux.tsx:251]: (same for above on line 132 and also in groupux.tsx)

File: brainstorm/src/react/buttonux.tsx [hunk 2]
```diff
@@ -15,26 +15,30 @@ import {
...
-
-		for (const id of ids) {
-			const n = findNote(props.items, id);
-			if (n instanceof Note) {
-				moveItem(n, Infinity, group.items);
+		// Wrap the add group operation in a transaction as it adds a group and potentially moves
+		// multiple notes into the group and we want to ensure that the operation is atomic.
+		// This ensures that the revertible of the operation will undo all the changes made by the operation.
+		Tree.runTransaction(props.items, () => {
+			e.stopPropagation();
```
Comment (ChumpChief) [inline brainstorm/src/react/buttonux.tsx:32]: Probably move the stopPropagation() call outside the transaction?  stopPropagation won't function if it isn't called during propagation, and it's not immediately clear if Tree will immediately invoke this callback or not.

**Review by ChumpChief — COMMENTED** [review]
Some cool stuff in here!

File: brainstorm/src/utils/undo.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+			revertible[disposeSymbol]();
+		}
+		for (const revertible of redoStack) {
+			revertible[disposeSymbol]();
+		}
+	};
+	return { undoStack, redoStack, unsubscribe };
+}
+
+export function revertFromStack(stack: Revertible[]): void {
```
[the remark this answers is not in the document]
Comment (nmsimons) [reply brainstorm/src/utils/undo.ts:62]: Yeah, I actually meant to make that change but ran out of time and wanted to be done. I will revisit.

File: brainstorm/src/react/noteux.tsx [hunk 4]
```diff
@@ -242,13 +240,18 @@ function NoteToolbar(props: { note: Note; clientId: string; notes: Items }): JSX
 
 export function AddNoteButton(props: { parent: Items; clientId: string }): JSX.Element {
 	const [{ isActive }, drop] = useDrop(() => ({
-		accept: dragType.NOTE,
+		accept: [dragType.NOTE, dragType.GROUP],
 		collect: (monitor) => ({
 			isActive: monitor.canDrop() && monitor.isOver(),
 		}),
+		canDrop: (item) => {
+			if (item instanceof Note) return true;
```
[the remark this answers is not in the document]
Comment (nmsimons) [reply brainstorm/src/react/noteux.tsx:248]: I can change this to Tree.is

File: brainstorm/src/react/noteux.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nmsimons) [reply brainstorm/src/react/noteux.tsx:251]: I can change these all to Tree.is

File: brainstorm/src/index.tsx [hunk 5]
```diff
@@ -41,11 +42,13 @@ async function start() {
 		initialContainers: [
 			{
 				container,
-				containerKey: "My Container",
+				containerKey: "main",
 			},
 		],
 	});
 
+	const undoRedoStacks = createUndoRedoStacks(appTree.events);
```
[the remark this answers is not in the document]
Comment (nmsimons) [reply brainstorm/src/index.tsx:50]: I wrote it that way initially but it ends up being a bit cumbersome as it requires useEffect and useState. By doing it outside of the react app, it is less awkward.

File: brainstorm/src/react/buttonux.tsx (hunk 2 above)
Comment (nmsimons) [reply brainstorm/src/react/buttonux.tsx:32]: Good call

**Review by ChumpChief — APPROVED** [review]

**Pull request merged**